sectionsUtilityTool/src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionUtilityToolFilter.java

```diff
@@ -0,0 +1,171 @@
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import edu.umich.its.lti.utils.PropertiesUtilities;
+
+
+
+public class SectionUtilityToolFilter implements Filter {
+	private static final String OU_GROUPS = "ou=Groups";
+	private static final String PROVIDER_URL = "ldap://ldap.itd.umich.edu:389/dc=umich,dc=edu";
+	private static final String LDAP_CTX_FACTORY = "com.sun.jndi.ldap.LdapCtxFactory";
+	private static final String ASSETS_PATH = "/assets";
+	private static Log M_log = LogFactory.getLog(SectionUtilityToolFilter.class);
+	protected static final String SYSTEM_PROPERTY_FILE_PATH = "sectionsToolPropsPath";
+	protected static final String PROPERTY_CANVAS_ADMIN = "canvas.admin.token";
+	protected static final String PROPERTY_CANVAS_URL = "canvas.url";
+	protected static final String PROPERTY_TEST_URL = "test.url";
+	private static final String AUTH_GROUP = "its-canvas-sections";
+	private static final String TEST_USER = "testUser";
+	protected static Properties canvasProperties = null;
+	private boolean isTestUrlEnabled=false;
+
+	@Override
+	public void init(FilterConfig filterConfig) throws ServletException {
+		M_log.debug("Filter Init(): Called");
+		getCanvasCredentials();
+		
+	}
+
+	@Override
+	public void doFilter(ServletRequest request, ServletResponse response,
+			FilterChain chain) throws IOException, ServletException {
+		M_log.debug("doFilter: Called");
+		HttpServletRequest useRequest = (HttpServletRequest) request;
+		HttpServletResponse useResponse=(HttpServletResponse)response;
+		if(!checkForAuthorization(useRequest)) {
+		      useResponse.sendError(403);
+			return;
+		}
+		chain.doFilter(useRequest, response);
+	}
+
+	@Override
+	public void destroy() {
+		M_log.debug("detroy: Called");
+	}
+	protected void getCanvasCredentials() {
+		M_log.debug("getCanvasCredentials(): called");
+		String propertiesFilePath = System.getProperty(SYSTEM_PROPERTY_FILE_PATH);
+		if (!isEmpty(propertiesFilePath)) {
+		canvasProperties=PropertiesUtilities.getPropertiesObjectFromURL(propertiesFilePath);
+		}else {
+			M_log.error("File path for (sectionsToolProps.properties) is not provided");
+		}
+		
+		
+	}
+	 private boolean isEmpty(String value) {
+		return (value == null) || (value.trim().equals(""));
+	}
+	
+	/*
+	 * User is authenicated using cosign  authorized using Ldap. For local development we are enabling
+	 * "testUser" parameter. "testUser" also go through the Ldap authorization process. 
+	 * we have test.url configured in the properties file in order to disable usage of "testUser" parameter in PROD.
+	 * "/assets" folder is where all the js/css resources. I am not making those folder go through authorization process for
+	 * 2 reasons 1) If the request of /index.html page fails we return the error page and that needs .css file for doing styling.
+	 * 2) if /index.html is 200 ok, the next request that follows are /assets. so it kind of redundant to go through ldap check.
```

Comment:
What will happen if ask for /assets without getting index.html?  Is it impossible to request /assets without going through index.html?